pkg/config/config.go

```diff
@@ -12,29 +12,88 @@ import (
 	"time"
 
 	"github.com/bacalhau-project/bacalhau/pkg/storage/util"
+	"github.com/bacalhau-project/bacalhau/pkg/util/filefs"
 	"github.com/libp2p/go-libp2p/core/crypto"
 	"github.com/rs/zerolog/log"
 )
 
 const (
-	maxUInt16 uint16 = 0xFFFF
-	minUInt16 uint16 = 0x0000
+	maxUInt16              uint16 = 0xFFFF
+	minUInt16              uint16 = 0x0000
+	DefaultRunInfoFilename        = "bacalhau.run"
+	RunInfoFilePermissions        = 0400
 )
 
-func DevstackShouldWriteEnvFile() bool {
-	return DevstackEnvFile() != ""
+var RunInfoFilePath = ""
+
+func DevstackGetShouldPrintInfo() bool {
+	return os.Getenv("DEVSTACK_PRINT_INFO") != ""
+}
+
+func DevstackSetShouldPrintInfo() {
+	os.Setenv("DEVSTACK_PRINT_INFO", "1")
 }
 
 func DevstackEnvFile() string {
 	return os.Getenv("DEVSTACK_ENV_FILE")
 }
 
-func DevstackGetShouldPrintInfo() bool {
-	return os.Getenv("DEVSTACK_PRINT_INFO") != ""
+// WriteRunInfoFile writes the ShellVariables to a file that can be imported. It writes bacalhauServe.run to TempDir().
+func WriteRunInfoFile(ctx context.Context, summaryShellVariablesString string) error {
+	writePath := ""
+	if writeable, _ := filefs.IsWritable("/run"); writeable {
+		writePath = "/run" // Linux
+	} else if writeable, _ := filefs.IsWritable("/var/run"); writeable {
+		writePath = "/var/run" // Older Linux
+	} else if writeable, _ := filefs.IsWritable("/private/var/run"); writeable {
+		writePath = "/private/var/run" // MacOS
```

Comment:
These folders all tend to be owned by `root` so we'll only be able to write to them if bacalhau is run as root.   On Linux we _can_ write to /var/run/user/[uid] if we know the current user's UID.  Otherwise if it is a file intended to be owned by a specific (non-root) user, then it should probably be in `~/username/.bacalhau`

Reply:
agreed - this is what "IsWriteable" checks. Here's a question - do we expect bacalhau to be run as non-root? I'd imagine so - i'm just not sure writing a global service to a user's directory is the best.

Reply:
We _should_ be able to run as non-root, but I'm not sure it's the case today. Either way ~/.bacalhau is probably not a useful place for info to be shared. If we default to /tmp/ when we can't write to any of the run directories, is that useful for something else trying to find it?

Reply:
my concern with /tmp is just it's not a very secure place (and information here will allow anyone to join a bacalhau network)